clang/lib/CodeGen/CGDeclCXX.cpp

```diff
@@ -1121,6 +1121,11 @@ CodeGenFunction::GenerateCXXGlobalInitFunc(llvm::Function *Fn,
       if (Decls[i])
         EmitRuntimeCall(Decls[i]);
 
+    if (getLangOpts().HLSL)
+      if (llvm::Function *ResInitFn =
+              CGM.getHLSLRuntime().createResourceBindingInitFn())
```

Comment:
Originally I asked: "What does it mean if `createResourceBindingInitFn` returns nullptr?" I asked this because I assumed that this function returning nullptr would indicate an error. I think it actually returns null if no function is needed? Wonder if this could be reflected in the name? Or maybe this is a common convention in this codebase?

Reply:
I could not figure out the good way to reflect it in the name, so I have added `needsResourceBindingInitFn` that can be used to check if the init function is needed, and `createResourceBindingInitFn` will always return non-nullptr.

Reply:
Yeah, I had `createResourceBindingInitFnIfNeeded` and did not like it. The `maybe*` variant is better, but I think I'll just leave this as is now.